diamond/diamond/attributewidget.py

```diff
@@ -105,13 +105,13 @@ def update(self, node):
         model.set_value(iter, 2, cell_model)
 
         if isinstance(node.attrs[key][0], tuple):
-          if node.attrs[key][1] is None:
+          if node.get_attr(key) is None:
```

Comment:
If I got it well, `node.get_attr(key)` is the same thing as `node.attrs[key]`? So in the previous version, the None comparison was not made on the correct object and that is why there was a problem? Meaning that if you write `node.attrs[key]` instead of `node.get_attr(key)`, you would still get the same expected correct behavior?

Reply:
It's been a while since I looked at this but I believe `node.attrs` contains a copy of the same data that `node.get_attr` is accessing yes but that copy only gets updated when the focus changes and the `MixedTree` object is reinitialized.  So you would not expect the same behavior from the two, no.  Or rather, you only get the same behavior when you switch focus between changes of the attribute value, which was the original bug.

I'm guessing that `node.get_attr` was previously not used because there was a bug in it that always returned `None` because it was missing `return` statements.  This is fixed here so that function becomes valid again.